### ISSUE TITLE
restrict users

### ISSUE THREAD

**Issue by tcurdt** [nowaythatworked/auth-astro#64]
Is there a way to use an OAuth provider and still restrict the user?

For example only allow certain users from Github?

I didn't find that yet.

**Comment by nowaythatworked** [comment]
Yes, you can define a callback function that handles extra authentication steps. For your use case the signIn callback is probably the correct one. https://authjs.dev/guides/basics/callbacks
